tests/models/nextflow/test_nextflow_deliver.py

```diff
@@ -1,45 +1,110 @@
-        deliverables=nextflow_deliverables
+    file_deliverable: FileDeliverable = FileDeliverable(
+        format=file_format,
+        id=deliverable_id,
+        path=existing_file,
+        step=step,
+        tag=step,
     )
 
     # THEN assert that it was successfully created
-    assert isinstance(nextflow_deliverables_object, NextflowDeliverables)
+    assert isinstance(file_deliverable, FileDeliverable)
+
+
+def test_file_deliverables_missing_mandatory(
+    deliverable_id: str,
+    step: str,
+    existing_file: Path,
+):
+    """Tests file delivery when a mandatory field is missing."""
+    # GIVEN valid deliverables fields
+
+    # WHEN instantiating a deliverables object
+    # THEN assert that an error is raised
+    with pytest.raises(PydanticValidationError) as error:
+        FileDeliverable(
+            id=deliverable_id,
+            path=existing_file,
+            step=step,
+            tag=step,
+        )
+    # THEN assert the error message
+    assert "field required (type=value_error.missing)" in str(error.value)
 
 
-def test_instantiate_nextflow_deliverables_with_empty_entry(
-    nextflow_deliverables_with_empty_entry: Dict[str, Optional[str]],
+def test_file_deliverables_non_existing_attribute(
+    deliverable_id: str, file_format: str, step: str, existing_file: Path
 ):
-    """
-    Tests nextflow delivery object with empty entry.
-    """
-    # WHEN instantiating a deliverables object with an empty entry THEN assert that it was successfully created
-    with pytest.raises(PydanticValidationError):
-        NextflowDeliverables(deliverables=nextflow_deliverables_with_empty_entry)
+    """Tests file delivery when a non existing attribute is given."""
+    # GIVEN valid deliverables fields
+
+    # WHEN instantiating a deliverables object
+    file_deliverable: FileDeliverable = FileDeliverable(
+        format=file_format,
+        id=deliverable_id,
+        path=existing_file,
+        step=step,
+        tag=step,
+        nonexisting=file_format,
+    )
+
+    # THEN assert that it was successfully created
+    assert isinstance(file_deliverable, FileDeliverable)
+
+    # THEN assert that not existing attribute is not included
+    assert "nonexisting" not in file_deliverable.__annotations__.keys()
 
 
-def test_instantiate_nextflow_deliverables_with_faulty_entry(
-    nextflow_deliverables_with_faulty_entry: Dict[str, str],
+def test_file_deliverables_non_existing_file(
+    deliverable_id: str,
+    file_format: str,
+    step: str,
+    non_existing_file: Path,
 ):
-    """
-    Tests nextflow delivery object with empty entry.
-    """
-    # WHEN instantiating a deliverables object with an empty entry THEN assert that it was successfully created
-    with pytest.raises(PydanticValidationError):
-        NextflowDeliverables(deliverables=nextflow_deliverables_with_faulty_entry)
+    """Tests file delivery when a mandatory file does not exist."""
+    # GIVEN valid deliverables fields
```

Comment:
I am maybe misunderstanding this a bit - what are valid deliverables fields?

Reply:
your understanding is great. the comments are not. changing!